ash/src/entry_libloading.rs

```diff
@@ -71,13 +83,25 @@ impl EntryCustom<Arc<Library>> {
     /// `dlopen`ing native libraries is inherently unsafe. The safety guidelines
     /// for [`Library::new`] and [`Library::get`] apply here.
     pub unsafe fn with_library(path: impl AsRef<OsStr>) -> Result<Entry, LoadingError> {
-        let lib = Library::new(path).map_err(LoadingError).map(Arc::new)?;
+        let lib = Library::new(path)
+            .map_err(LoadingError::LibraryLoadFailure)
+            .map(Arc::new)?;
 
-        Ok(Self::new_custom(lib, |vk_lib, name| {
+        let mut library_name = None;
+        let result = Self::new_custom(lib, |vk_lib, name| {
             vk_lib
                 .get(name.to_bytes_with_nul())
                 .map(|symbol| *symbol)
-                .unwrap_or(ptr::null_mut())
-        }))
+                .unwrap_or_else(|_| {
```

Comment:
I don't really like loosing error information here. Since you're extending `LoadingError` with a case specific to symbol loading, perhaps keep `libloading::Error` inside the `MissingEntryPoint` variant in addition to the name.